### PR TITLE
Third part

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

File: src/main/java/mate/academy/util/ConnectionUtil.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Can't find MySQL Driver", e);
+        }
+    }
+
+    public static Connection getConnection() {
+        Properties dbProperties = new Properties();
+        dbProperties.put("user", "root");
+        dbProperties.put("password", "1234");
```
Comment (vlykhvar) [inline src/main/java/mate/academy/util/ConnectionUtil.java]: i think better to make variables for login and password values

File: src/main/java/mate/academy/Main.java [hunk 2]
```diff
@@ -108,4 +110,15 @@ public static void main(String[] args) {
         carByDriverKate.forEach(System.out::println);
         carByDriverPetro.forEach(System.out::println);
     }
+
+    private static void truncate() {
```
Comment (vlykhvar) [inline src/main/java/mate/academy/Main.java]: The method should not be placed in main.class.

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (manufacturer_name, manufacturer_country)"
+                + " VALUES (?,?);";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            return manufacturer;
```
Comment (vlykhvar) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java:34]: manufacturer what is returned has not have an id as i understand. We don't have able to use them next.

**Review by vlykhvar — CHANGES_REQUESTED** [review]

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+@Dao
+public class ManufacturerDaoJdbc implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (manufacturer_name, manufacturer_country)"
+                + " VALUES (?,?);";
+        String querySelectAll = "SELECT * FROM manufacturers;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java]:
```suggestion
                PreparedStatement statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```
So you will be able to use: 
statement.executeUpdate();
            ResultSet resultSet = statement.getGeneratedKeys();

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create manufacturer: " + manufacturer, e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers WHERE manufacturer_id = ?;";
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java]:
```suggestion
        String getByIdQuery = "SELECT * FROM manufacturers WHERE manufacturer_id = ?;";
```
make your string a bit more informative

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long manufactureId = resultSet.getObject("manufacturer_id", Long.class);
+                String manufacturerName = resultSet.getObject("manufacturer_name", String.class);
+                String manufacturerCountry =
+                        resultSet.getObject("manufacturer_country", String.class);
+                manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+                manufacturer.setId(manufactureId);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java]: let's move this logic to a private method and you will be able to reuse it in getAll() method

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java (hunk 5 above)
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java]: do we want to get deleted manufcaturers as well?

File: src/main/resources/init_db.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
+CREATE SCHEMA `taxi_service` DEFAULT CHARACTER SET utf8 ;
+CREATE TABLE `taxi_service`.`manufacturers` (
+    `manufacturer_id` BIGINT NOT NULL AUTO_INCREMENT,
+    `manufacturer_name` VARCHAR(225) NOT NULL,
+    `manufacturer_country` VARCHAR(225) NOT NULL,
+    `is_deleted` BIT NOT NULL DEFAULT 0,
+PRIMARY KEY (`manufacturer_id`));
```
Comment (RomanDubovskyi) [inline src/main/resources/init_db.sql]: add an empty line at the end of the file

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    public List<Manufacturer> getAll() {
+        String getAllQuery = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = getManufacturer(resultSet);
+                manufacturers.add(manufacturer);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/jdbc/ManufacturerDaoJdbc.java:69]:
```suggestion
                manufacturers.add(getManufacturer(resultSet));
```

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**